ParquetSharp.Dataset/TypeComparer.cs

```diff
@@ -0,0 +1,196 @@
+using Apache.Arrow;
+using Apache.Arrow.Types;
+
+namespace ParquetSharp.Dataset;
+
+internal sealed class TypeComparer
```

Comment:
There's similar functionality in the Arrow .NET library but it's internal. It would be nice to make that public so we could get rid of this later.